lib/model/unittest/CMetricModelTest.cc

```diff
@@ -2292,44 +2292,54 @@ BOOST_FIXTURE_TEST_CASE(testIgnoreSamplingGivenDetectionRules, CTestFixture) {
     for (std::size_t i = 0; i < 60; i++) {
+        this->addArrival(SMessage(startTime + i, "p1", 110.0), gathererNoSkip);
         this->addArrival(SMessage(startTime + i, "p1", 110.0), gathererWithSkip);
     }
+    modelNoSkip->sample(startTime, endTime, m_ResourceMonitor);
     modelWithSkip->sample(startTime, endTime, m_ResourceMonitor);
 
+    // Checksums will be different due to the small weighting applied to the sample
+    // added to the model with the detector rule.
+    BOOST_TEST_REQUIRE(modelWithSkip->checksum() != modelNoSkip->checksum());
+
     startTime = endTime;
     endTime += bucketLength;
 
-    // Wind the other model forward
-    modelNoSkip->skipSampling(startTime);
-
+    // Add more data to both models, for which the detection rule will not apply
     for (std::size_t i = 0; i < 60; i++) {
         this->addArrival(SMessage(startTime + i, "p1", 2.0), gathererNoSkip);
         this->addArrival(SMessage(startTime + i, "p1", 2.0), gathererWithSkip);
     }
     modelNoSkip->sample(startTime, endTime, m_ResourceMonitor);
     modelWithSkip->sample(startTime, endTime, m_ResourceMonitor);
 
-    // Checksums will be different due to the data gatherers
+    // Checksums will be different due to the small weighting applied to the sample
+    // added to the model with the detector rule.
     BOOST_TEST_REQUIRE(modelWithSkip->checksum() != modelNoSkip->checksum());
 
-    // TODO this test fails due a different checksums for the decay rate and prior
-    // but the underlying models should be the same.
-    // See elastic/ml-cpp/issues/2043
-    // CAnomalyDetectorModel::TModelDetailsViewUPtr modelWithSkipView =
-    //     modelWithSkip->details();
-    // CAnomalyDetectorModel::TModelDetailsViewUPtr modelNoSkipView = modelNoSkip->details();
-
-    // uint64_t withSkipChecksum = modelWithSkipView->model(model_t::E_IndividualMeanByPerson, 0)->checksum();
-    // uint64_t noSkipChecksum = modelNoSkipView->model(model_t::E_IndividualMeanByPerson, 0)->checksum();
-    // BOOST_REQUIRE_EQUAL(withSkipChecksum, noSkipChecksum);
+    // The underlying models should also differ due to the different weighting applied to the samples.
+    CAnomalyDetectorModel::TModelDetailsViewUPtr modelWithSkipView =
+        modelWithSkip->details();
+    CAnomalyDetectorModel::TModelDetailsViewUPtr modelNoSkipView = modelNoSkip->details();
+
+    const maths::CModel* mathsModelWithSkip =
+        modelWithSkipView->model(model_t::E_IndividualMeanByPerson, 0);
+    BOOST_TEST_REQUIRE(mathsModelWithSkip != nullptr);
+    uint64_t withSkipChecksum = mathsModelWithSkip->checksum();
+    const maths::CModel* mathsModelNoSkip =
+        modelNoSkipView->model(model_t::E_IndividualMeanByPerson, 0);
+    BOOST_TEST_REQUIRE(mathsModelNoSkip != nullptr);
+    uint64_t noSkipChecksum = mathsModelNoSkip->checksum();
+    BOOST_TEST_REQUIRE(withSkipChecksum != noSkipChecksum);
 
     // TODO These checks fail see elastic/ml-cpp/issues/2043
```

Comment:
This issue doesn't seem to tie up with the code that's commented out.  The issue says the problem is with checksums, but the code below is comparing last times.

It seems like we shouldn't have this issue reference here and either the last times _should_ be the same, or, if they are not then it's because we've skipped data from the most recent bucket and the commented out assertions should just be deleted as they're obviously never going to work.